src/status_im2/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -47,71 +48,88 @@
-        current-limit         (reagent/atom {:amount   limit-crypto
-                                             :currency token})
-        handle-swap           (fn [crypto?]
-                                (let [num-value (parse-double @input-value)]
-                                  (reset! current-limit (if crypto?
-                                                          {:amount   limit-crypto
-                                                           :currency token}
-                                                          {:amount   limit-fiat
-                                                           :currency currency}))
-                                  (when (> num-value (:amount @current-limit))
-                                    (reset! input-value ""))))
-        handle-keyboard-press (fn [v]
-                                (let [current-value @input-value
-                                      new-value     (make-new-input current-value v)
-                                      num-value     (or (parse-double new-value) 0)]
-                                  (when (<= num-value (:amount @current-limit))
-                                    (reset! input-value new-value)
-                                    (reagent/flush))))
-        handle-delete         (fn [_]
-                                (swap! input-value #(subs % 0 (dec (count %))))
-                                (reagent/flush))
-        handle-on-change      (fn [v]
-                                (when (valid-input? @input-value v)
-                                  (let [num-value            (or (parse-double v) 0)
-                                        current-limit-amount (:amount @current-limit)]
-                                    (if (> num-value current-limit-amount)
-                                      (reset! input-value (str current-limit-amount))
-                                      (reset! input-value v))
-                                    (reagent/flush))))]
+  [{:keys [rate]}]
+  (let [bottom                    (safe-area/get-bottom)
+        {:keys [currency]}        (rf/sub [:profile/profile])
+        networks                  (rf/sub [:wallet/network-details])
+        wallet-send               (get-in (rf/sub [:wallet]) [:ui :send])
+        token                     (:token wallet-send)
+        token-symbol              (:symbol token)
+        limit-crypto              (:total-balance token)
+        limit-fiat                (:total-balance-fiat token)
+        conversion-rate           (or rate 10)
+        input-value               (reagent/atom "")
+        current-limit             (reagent/atom {:amount   limit-crypto
+                                                 :currency token-symbol})
+        loading-suggested-routes? (:loading-suggested-routes? wallet-send)
+        handle-swap               (fn [crypto?]
+                                    (let [num-value (parse-double @input-value)]
```

Comment:
parse-double is okay to use here? 
cc @ulisesmac

Reply:
I had no major issues while testing, but I agree we should use a safer approach using bignumbers. I made no changes to this code, so I would prefer to handle this in a separate issue! I will create it